Mathlib/Algebra/PUnitInstances.lean

```diff
@@ -69,6 +69,12 @@ theorem inv_eq : x⁻¹ = unit :=
 #align punit.inv_eq PUnit.inv_eq
 #align punit.neg_eq PUnit.neg_eq
 
+@[to_additive] instance [One M] : Subsingleton (OneHom PUnit M) where
+  allEq f g := FunLike.ext _ _ <| Unique.forall_iff.2 <| (map_one f).trans (map_one g).symm
```

Comment:
Shouldn't these instead be defined for some arbitrary `Unique` domain?

Reply:
I'm trying to add a very generic instance. Possibly, I shouldn't do that because it will try to generate `[OneHomClass _ _ _]` every time Lean is looking for `[Subsingleton _]`.

Reply:
I would recommend sticking to the concrete `OneHom` spelling.